forum_leaderboard/static/js/src/forum_leaderboard_studio.js

```diff
@@ -0,0 +1,21 @@
+/* Javascript for ForumLeaderboardXBlock. */
+function ForumLeaderboardStudioXBlock(runtime, element) {
+    $('.save-button', element).bind('click', function() {
+        var handlerUrl = runtime.handlerUrl(element, 'studio_submit');
+        var data = {
+            discussion_id: $(element).find('input[name=discussion_id]').val(),
+            count: $(element).find('input[name=count]').val()
+        };
+        $.post(handlerUrl, JSON.stringify(data)).success(function(response) {
+            if (response['success']) {
+                window.location.reload(false);
+                return
+            }
+            alert(response['errors'].join('\n'));
```

Comment:
@Kelketek @e-kolpakov While I was working on the new grade leaderboard I found out that there is a better way to do this for XBlocks, beyond what I already posted:

1. XBlocks do have their own validation system. It's apparently undocumented (except [in the XBlock source code](https://github.com/edx/XBlock/blob/master/xblock/validation.py)) but seems to work well, and the Studio validation just extends it. [Example from the grade leaderboard block](https://github.com/open-craft/xblock-leaderboard/commit/2eb30e85a0aa5fdf60700723fdf6335d54218f55#diff-e5c97498d81b78b05ef00be27015a3e8R41). You can't really use this when saving changes to the block, but can it makes it less necessary to check some fields during save.

2. Remember our earlier discussion about `location.reload`? There is a much nicer way to do this: [using runtime.notify("save")](https://github.com/open-craft/xblock-leaderboard/commit/2eb30e85a0aa5fdf60700723fdf6335d54218f55#diff-ce688558df72d16b882bbfdccc3711f9R10). That way avoids reloading the page, displays nice "Saving..." animations, and even contains a nice display of any errors if the current AJAX request returns an HTTP error code.

Jonathan, if you want to leave things as they are now and merge this PR, I can make those changes on my branch as part of my refactoring. If not, feel free to make them now and I'll try to rebase my work onto it.